### ISSUE TITLE
Does not recognize shared databases on the WSL2 network

### ISSUE THREAD

**Issue by alvaroico** [beekeeper-studio/beekeeper-studio#387]
** Describe the bug **
The program cannot read the file shared on the WSL2 network example path: "\\ wsl $ \ Ubuntu \ home \ alvaroico \ GitHub \ NextLevelWeek3 \ backend \ src \ database \ database.sqlite"
Qaundo accesses the file through the WSL2 Network and registers that the file is empty

** Reproduce **
1. Run a project in WSL2
2. point to the WSL2 network path
3. example path "\\ wsl $ \ Ubuntu \ home \ alvaroico \ GitHub \ NextLevelWeek3 \ backend \ src \ database \ database.sqlite"
4. The application shows that the file was read but informs that the file is empty

** Expected behavior **
Normal reading of the SQLite file

** Screenshots / Gifs **
![image](https://user-images.githubusercontent.com/59286021/95936917-089de180-0dad-11eb-8122-2667ab932e04.png)
![image](https://user-images.githubusercontent.com/59286021/95936954-16536700-0dad-11eb-84e3-9e5a2022fc44.png)

** Version information (fill in the following information): **
 - OS: Microsoft Windows [version 10.0.19042.572]
 - Beekeeper Studio 1.7.5 application version

** Additional context **
If I copy the file to my computer locally it is read normally but when it is on the WSL2 network it is read according to the file it is empty

* Text made by Google Translate

**Comment by rathboma** [comment]
Hey! I'm a big WSL user, and I also have issues accessing databases across the WSL boundry. Good bug report. Not sure if this is an issue with Beekeeper Studio or Microsoft honestly.

I also have issues accessing PSQL running on WSL using localhost.

A workaround is to install the Linux version of Beekeeper Studio and use that!

**Comment by rathboma** [comment]
@alvaroico can you try it with forward slashes?
```
//wsl$/Ubuntu/home/alvaroico/GitHub/NextLevelWeek3/backend/src/database/database.sqlite
```

**Comment by alvaroico** [comment]
> Hey! I'm a big WSL user, and I also have issues accessing databases across the WSL boundry. Good bug report. Not sure if this is an issue with Beekeeper Studio or Microsoft honestly.
> 
> I also have issues accessing PSQL running on WSL using localhost.
> 
> A workaround is to install the Linux version of Beekeeper Studio and use that!
> 
> 

That's right I also have the same problem just the file is pointing to the network.
And running the file locally on WSL2 causes a redundancy of having 2 programs installed on the machine

**Comment by alvaroico** [comment]
> @alvaroico can you try it with forward slashes?
> ```
> //wsl$/Ubuntu/home/alvaroico/GitHub/NextLevelWeek3/backend/src/database/database.sqlite
> ```

This is the official WSL2 path as per Microsoft documentation

https://docs.microsoft.com/pt-br/windows/wsl/compare-versions

**Comment by rathboma** [comment]
Yep. I'll figure it out eventually. My gut is that this is some weird behavior due to how WSL works, but probably something we can do on our side.

**Comment by pgThiago** [comment]
Eu com mesmo BO aqui na NLW-5 haha.

Same problem while doing NLW-5 hehe.

**Comment by mbthales** [comment]
Any solution for this problem?

**Comment by alvaroico** [comment]
This issue was resolved on windows 11 by installing a browser on WSL2 to open

**Comment by alvaroico** [comment]
> > This issue was resolved on windows 11 by installing a browser on WSL2 to open
> 
> Hey @alvaroico I'm facing this issue in W11 but I'm not sure what are you referring to in your comment to fix it.
> 
> Could you please expand a little so I can follow?
> 
> Appreciate your help 😄

You have to install it in WSL2 so that it has direct access to the file without the network problem that you have using reading outside of WSL2

**Comment by jaymgonzalez** [comment]
Cool! Thanks for the help, it worked!

**Comment by alvaroico** [comment]
@rathboma an update on the bug in networked WSL2. PT-BR: uma atualização sobre o erro no WSL2 em rede.
![erro](https://github.com/beekeeper-studio/beekeeper-studio/assets/59286021/ae42a6dd-d0b2-4381-a4f8-5d3fcbdd0b80)

**Comment by eriket0107** [comment]
@alvaroico hey man! did you had any news about this issue? I'm facing it right now 
fala mano! você teve alguma notícia sobre esse problema? to passando por isso agora

**Comment by alvaroico** [comment]
@eriket0107 
I made a tutorial to use on Windows 11
![beekeeperWin11](https://github.com/beekeeper-studio/beekeeper-studio/assets/59286021/72d888eb-cb88-4525-9561-8e07ae14f22b)

**Comment by bhuynhdev** [comment]
Seems like this is Windows issue, not BeeKeeper. I have tried with HeidiSQL and same problem

DBeaver is reportedly also having same problem https://github.com/dbeaver/dbeaver/issues/17217

Maybe the BeeKeeper team can have some instructions on how to install BeeKeepr on WSL2 now that it has GUI supprt

**Comment by colinmollenhour** [comment]
Unfortunately, running GUIs from WSL2 is pretty bad and will never be "great". Mainly the scaling is the issue, but you lose a lot of the niceties of normal Windows with WSL2 apps and they crash often, performance is bad, etc.

What would be ideal is special support for the WSL2 filesystem, similar to how VS Code supports WSL2 filesystem from running VS Code in Windows. For example, it may require launching a lightweight "connector" app inside of the WSL environment that opens the file, then Beekeeper talks to this connector through a TCP port.

This would set Beekeeper apart from all competitors since AFAIK none of them support opening files over the WSL2 file system (errors with locking the database).
